test/PowerShellEditorServices.Test/Debugging/DebugServiceTests.cs

```diff
@@ -2,101 +2,141 @@
-using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
 using System.Management.Automation;
 using System.Threading;
 using System.Threading.Tasks;
-using MediatR;
 using Microsoft.Extensions.Logging.Abstractions;
 using Microsoft.PowerShell.EditorServices.Services;
 using Microsoft.PowerShell.EditorServices.Services.DebugAdapter;
-using Microsoft.PowerShell.EditorServices.Services.PowerShell;
 using Microsoft.PowerShell.EditorServices.Services.PowerShell.Host;
 using Microsoft.PowerShell.EditorServices.Services.TextDocument;
 using Microsoft.PowerShell.EditorServices.Test.Shared;
-using Newtonsoft.Json.Linq;
-using OmniSharp.Extensions.JsonRpc;
-using OmniSharp.Extensions.LanguageServer.Protocol.Models;
-using OmniSharp.Extensions.LanguageServer.Protocol.Progress;
-using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using Microsoft.PowerShell.EditorServices.Utility;
 using Xunit;
-
 namespace Microsoft.PowerShell.EditorServices.Test.Debugging
 {
-    /*
     public class DebugServiceTests : IDisposable
     {
-        private WorkspaceService workspace;
-        private DebugService debugService;
-        private ScriptFile debugScriptFile;
-        private ScriptFile variableScriptFile;
+        private readonly PsesInternalHost psesHost;
+        private readonly BreakpointService breakpointService;
+        private readonly DebugService debugService;
+        private readonly BlockingCollection<DebuggerStoppedEventArgs> debuggerStoppedQueue = new();
+        private readonly WorkspaceService workspace;
+        private readonly ScriptFile debugScriptFile;
+        private readonly ScriptFile variableScriptFile;
+
+        public DebugServiceTests()
+        {
+            psesHost = PsesHostFactory.Create(NullLoggerFactory.Instance);
+            // This is required for remote debugging, but we call it here to end up in the same
+            // state as the usual startup path.
+            psesHost.DebugContext.EnableDebugMode();
+
+            breakpointService = new BreakpointService(
+                NullLoggerFactory.Instance,
+                psesHost,
+                psesHost,
+                new DebugStateService());
+
+            debugService = new DebugService(
+                psesHost,
+                psesHost.DebugContext,
+                remoteFileManager: null,
+                breakpointService,
+                psesHost,
+                NullLoggerFactory.Instance);
+
+            debugService.DebuggerStopped += OnDebuggerStopped;
+
+            // Load the test debug files
+            workspace = new WorkspaceService(NullLoggerFactory.Instance);
+            debugScriptFile = GetDebugScript("DebugTest.ps1");
+            variableScriptFile = GetDebugScript("VariableTest.ps1");
+        }
+
+        public void Dispose()
+        {
+            debugService.Abort();
+            psesHost.StopAsync().GetAwaiter().GetResult();
+            GC.SuppressFinalize(this);
```

Comment:
Try `sealed` class to see if warning goes away.

Reply:
@daxian-dbw Ah ha, xUnit will not allow the unit tests' classes to be `sealed`, they _must_ be `public` and you can't do `public sealed` (or `sealed public`). Oh well.